lib/api/multiObjectDelete.js

```diff
@@ -50,18 +53,27 @@ import { isBucketAuthorized } from './apiUtils/authorization/aclChecks';
 * @param {boolean} quietSetting - true if xml should just include error list
 * and false if should include deleted list and error list
 * @param {object []} errorResults - list of error result objects with each
-* object containing -- key: objectName, error: arsenal error
-* @param {string []} deleted - list of object keys deleted
+* object containing -- entry: { key, versionId }, error: arsenal error
+* @param {object []} deleted - list of object deleted, an object has the format
+*     object: { entry, result, isDeletingDeleteMarker }
+*     object.entry : above
+*     object.result: stringification of { versionId }
+*     object.isDeletingDeleteMarker: name as comment
 * @return {string} xml string
 */
 function _formatXML(quietSetting, errorResults, deleted) {
     let errorXML = [];
     errorResults.forEach(errorObj => {
         errorXML.push(
         '<Error>',
-        '<Key>', escapeForXML(errorObj.key), '</Key>',
-        '<Code>', errorObj.error.message, '</Code>',
-        '<Message>', errorObj.error.description, '</Message>',
+        '<Key>', escapeForXML(errorObj.entry.key), '</Key>',
+        '<Code>', errorObj.error.message, '</Code>');
+        if (errorObj.entry.versionId) {
+            const version = errorObj.entry.versionId === 'null' ?
+                'null' : escapeForXML(errorObj.entry.versionId);
```

Comment:
@alexandre-merle I think they are only encrypted if the error happens in _parseXml when decrypting, otherwise the versionId that is saved in the entry is the decrypted id. I'm thinking to save the original version ID in the entry as well so we don't have to re-encrypt ourselves.